### PR TITLE
Add `ActorControlSelfExtra`, additional categories

### PR DESCRIPTION
**Pull request by valarnin**
~~I haven't actually tested this PR yet, so it's being marked as draft for now.~~

Per discussion in https://github.com/OverlayPlugin/cactbot/pull/99, add some additional `ActorControl` handling.
Also some minor cleanup for `NetworkParser` where it was using a raw value for category.

New categories, plus previous `SetAnimationState` category for comparison:

| Count | Opcode | Category |
| - | - | - |
| 13 | ActorControl | DisplayPublicContentTextMessage |
| 54 | ActorControlSelf | DisplayLogMessage |
| 12121 | ActorControl | SetAnimationState |

---

I seem to vaguely recall discussion about adding `EObjSetState` as well but I can't track down that conversation to confirm. If we still want to move forward with that it's easy enough to add to this PR.

| Count | Packet Opcode | Category |
| - | - | - |
| 6616 | ActorControl | DirectorEObjMod |
| 5425 | ActorControl | EObjAnimation |
| 125839 | ActorControl | EObjSetParent |
| 8377 | ActorControl | EObjSetState |
| 12 | ActorControlSelf | EObjSetState |

`ActorControlTarget` was considered as well, but I only have four categories for that, none of which seem relevant to add:

|  Count | Category |
 | --  | --  |
   | 2484 | Emote |
 |  23556 | SetPlayerTarget |
  |247093 | SetTarget |
 |   1065 | SetTargetSign |

Total packets checked to get the numbers above: 39580631

### CODE REVIEW

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlCommon.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace RainbowMage.OverlayPlugin.NetworkProcessors
+{
+    public enum Server_ActorControlCategory : ushort
+    {
+        SetAnimationState = 0x003E, // 62
+        StatusUpdate = 0x01F8, // 504
+        // Name is a guess
+        DisplayLogMessage = 0x020F, // 527
+        DisplayLogMessageParams = 0x0210, // 528
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorControlCommon.cs:9]: What's included in this category?  I see it in `AllowedActorControlCategories` below, but I don't recall whether this was something we talked about (and I'm just forgetting, entirely likely) or if this is something new you found?

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlExtra.cs [hunk 2]
```diff
@@ -18,10 +19,10 @@ public class LineActorControlExtra
...
         private readonly FFXIVRepository ffxiv;
 
-        // Any category defined in this enum will be allowed as an emitted line
-        public enum Server_ActorControlCategory : ushort
-        {
-            SetAnimationState = 62,
+        // Any category defined in this array will be allowed as an emitted line
+        public static readonly Server_ActorControlCategory[] AllowedActorControlCategories = {
+            Server_ActorControlCategory.SetAnimationState,
+            Server_ActorControlCategory.DisplayPublicContentTextMessage
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorControlExtra.cs:25]: I like the idea of defining these categories in a single place like `LineActorControlCommon` - but would it make sense to also define the list of allowed categories for each type of `ActorControl` line in there too?  Then adding future categories would only require changing the one file.

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlSelfExtra.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+// The easiest place to test SetAnimationState is Lunar Subteranne.
+// On the first Ruinous Confluence, each staff has this line:
+// 273|2023-12-05T10:57:43.4770000-08:00|4000A145|003E|00000001|00000000|00000000|00000000|06e7eff4a949812c
+// On the second Ruinous Confluence, each staff has this line:
+// 273|2023-12-05T10:58:00.3460000-08:00|4000A144|003E|00000001|00000001|00000000|00000000|a4af9f90928636a3
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorControlSelfExtra.cs:12]: Probably can remove these comments, since they apply to `ActorControlExtra`.

**Review by wexxlee — COMMENTED** [review]
> I seem to vaguely recall discussion about adding EObjSetState as well but I can't track down that conversation to confirm. If we still want to move forward with that it's easy enough to add to this PR.

I'm not recalling it offhand, and I can't seem to track it down either.  I'd say let's leave it out for now - should be easy enough to add later if we remember.

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlCommon.cs (hunk 1 above)
Comment (valarnin) [reply OverlayPlugin.Core/NetworkProcessors/LineActorControlCommon.cs:9]: I added an additional column to `LineActorControlSelfExtra` to explain why this is included:
```cs
            // Some `LogMessage` messages can be triggered by both 0x020F and 0x0210 categories, not sure what the difference is
            // except that 0x0210 messages usually have another actor ID in the parameters
```

Hopefully that makes sense?

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlExtra.cs (hunk 2 above)
Comment (valarnin) [reply OverlayPlugin.Core/NetworkProcessors/LineActorControlExtra.cs:25]: I think I'd prefer to keep the allowed categories on their associated class from a code organization perspective.

Additionally, we'd have to have a static class def to attach them to in the `Common` file, we can't define a variable at the namespace level in C#.

**Review by wexxlee — APPROVED** [review]

**Pull request merged**